### PR TITLE
45 auth session flow

### PR DESCRIPTION
**Pull request by theacheng**

### CODE REVIEW

File: lib/services/validators/password_validator.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import 'package:vtenh/constants/format_constant.dart';
+import 'package:vtenh/services/validators/base_validator.dart';
+
+class PasswordValidator extends BaseValidator {
```
Comment (bookmebus) [inline lib/services/validators/password_validator.dart:4]: Most languages provide a strong password check lib. there are more complicated than this like entropy level. No need to have this one but use a third party instead

File: test/services/validators/email_validator_test.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:vtenh/services/validators/email_validator.dart';
+
+void main() {
+  group("EmailValidator", () {
+    test("return true if email is valid", () {
```
Comment (bookmebus) [inline test/services/validators/email_validator_test.dart:6]: add another test for failure case

File: test/services/validators/phone_number_validator_test.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:intl_phone_number_input/intl_phone_number_input.dart';
+import 'package:vtenh/services/validators/phone_number_validator.dart';
+
+void main() {
+  group("PhoneNumberValidator", () {
+    test("return true if phone number is valid!", () async {
```
Comment (bookmebus) [inline test/services/validators/phone_number_validator_test.dart:7]: add another test for failure case

File: test/services/validators/username_validator_test.dart [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:vtenh/services/validators/username_validator.dart';
+
+void main() {
+  group("UsernameValidator", () {
+    test("return true if username is not contain special char and start from 0", () {
```
Comment (bookmebus) [inline test/services/validators/username_validator_test.dart:6]: add another test for failure

**Review by bookmebus — CHANGES_REQUESTED** [review]

**Comment by theacheng** [comment]
Now at 3 PM pull request, I almost did it at 1 PM, then I face with a problem that I tried to search on google and found this: 
```
git clean -xfd
git stash save --keep-index
git stash drop
git pull
flutter doctor
```
This just clears all my changes and also file on git ignore even api_constrant.dart that I have to rewrite it.

File: lib/screens/authentication/components/social_sign_in.dart [hunk 5]
```diff
@@ -1,17 +1,10 @@
 import 'package:flutter/material.dart';
```
Comment (theacheng) [inline lib/screens/authentication/components/social_sign_in.dart:1]: Can't rename the file to social_signin.dart, when I saved it, it automatically renames to social_signIn.dart

**Pull request merged**